### ISSUE TITLE
[Bug]: non-ascii property name may lead an invalid js output

### ISSUE THREAD

**Issue by xc2** [web-infra-dev/rsbuild#1221]
### Version

```sh
System:
    OS: macOS 14.2.1
    CPU: (12) arm64 Apple M2 Max
    Memory: 75.66 MB / 32.00 GB
    Shell: 3.6.4 - /opt/homebrew/bin/fish
  Browsers:
    Chrome: 120.0.6099.199
    Safari: 17.2.1
  npmPackages:
    @rsbuild/test-helper: workspace:* => 0.2.18 
    @rsbuild/tsconfig: workspace:* => undefined
```

### Details

upstream issue: https://github.com/swc-project/swc/issues/8491

since rsbuild has swc's `asciiOnly` option enabled by default (output charset is set to `ascii` by default), this bug may cause inconvenience for some users.

I believe we should disable the `asciiOnly` option, which will be a breaking change, for swc minimizer  until swc resolves the issue. or just include it in the FAQ.

### Reproduce link

https://codesandbox.io/p/devbox/mc3s9g?file=/dist/static/js/index.65230c7f.js:1,52-1,67

### Reproduce Steps

1. yarn build
2. check dist/static/js/index.*.js, there's `{a\xf3b:"\xf3"}` in the code which is invalid

**Comment by chenjiahan** [comment]
Good news, SWC will fix it soon 😄 https://github.com/swc-project/swc/pull/8493

**Comment by xc2** [comment]
> Good news, SWC will fix it soon 😄 [swc-project/swc#8493](https://github.com/swc-project/swc/pull/8493)

Such productivity...

**Comment by chenjiahan** [comment]
Wait for Rspack / SWC new releases 😄
